### ISSUE TITLE
Error: TypeError: Cannot read property 'call' of undefined

### ISSUE THREAD

**Issue by ianhirschfeld** [JoshuaWalsh/gatsby-plugin-excerpts#2]
I'm trying to use this excerpt on html that is coming from markdown within yaml.

Plugins I'm using:
- `gatsby-transformer-yaml-full`
- `gatsby-yaml-full-markdown`
- `gatsby-plugin-excerpts`

Here is an example yaml file:
```yaml
name: Dropbox
description: !markdown |
  **Wherever you are**  
  Put your stuff in Dropbox and get to it from your computers, phones, or tablets. Edit docs, automatically add photos, and show off videos from anywhere.
```

My excerpts config looks like this:
```js
{
  resolve: `gatsby-plugin-excerpts`,
  options: {
    sources: {
      snippetBlocks: {
        type: 'htmlQuery',
        sourceField: 'description',
        stripSelector: 'a',
        elementReplacements: [
          {
            selector: 'h6',
            replaceWith: 'strong'
          },
          {
            selector: 'h5',
            replaceWith: 'h6'
          },
          {
            selector: 'h4',
            replaceWith: 'h5'
          },
          {
            selector: 'h3',
            replaceWith: 'h4'
          },
          {
            selector: 'h2',
            replaceWith: 'h3'
          },
        ],
      },
      default: {
        type: 'htmlQuery',
        sourceField: 'html',
        excerptSelector: 'html > *',
        ignoreSelector: 'img, .gatsby-highlight',
        stripSelector: 'a',
        elementReplacements: [
          {
            selector: 'h6',
            replaceWith: 'strong'
          },
          {
            selector: 'h5',
            replaceWith: 'h6'
          },
          {
            selector: 'h4',
            replaceWith: 'h5'
          },
          {
            selector: 'h3',
            replaceWith: 'h4'
          },
          {
            selector: 'h2',
            replaceWith: 'h3'
          },
        ],
        truncate: {
          length: 80,
          byWords: true,
          ellipsis: '…'
        },
      }
    },
    sourceSets: {
      markdownHtml: [
        `snippetBlocks`,
        `default`
      ]
    },
    excerpts: {
      snippet: {
        type: `html`,
        nodeTypeSourceSet: {
          ToolkitYaml: `markdownHtml`
        }
      }
    }
  },
}
```

When running this in the GraphQL explorer my console is outputting:
```
warn [gatsby-plugin-excerpts] Failed to retrieve field name for source snippetBlocks on node
{"name":"Dropbox","description":"<p><strong>Wherever you are</strong><br>\nPut your stuff in Dropbox and get to it
from your computers, phones, or tablets. Edit docs, automatically add photos, and show off videos from
anywhere.</p>\n<p><strong>Share with confidence</strong><br>\nShare photos with friends. Work with your team like
you're using a single computer. Everything's automatically private, so you control who sees what.</p>\n<p><strong>Safe
 and secure</strong><br>\nEven if your phone goes for a swim, your stuff is always safe in Dropbox and can be restored
 in a snap. Dropbox secures your files with 256-bit AES encryption and two-step verification.</p>\n<p><strong>Dropbox
for Business</strong><br>\nMillions of people use Dropbox at work. With Dropbox for Business, get the power and
security of Dropbox plus robust admin controls, dedicated support, and all the space you need.</p>\n","id":"69da20e4-3
9cf-5a8e-942e-82bd8bc7aa6a","parent":"6d788dd1-4504-5c96-b52b-79dbf63c0954","children":[],"internal":{"contentDigest":
"a307bad168cc338874d1930aa890d197","type":"ToolkitYaml","counter":40,"owner":"gatsby-transformer-yaml-full"}} . Error:
  TypeError: Cannot read property 'call' of undefined
    at /Users/hirsch/Development/hackdesign-gatsby/node_modules/gatsby-plugin-excerpts/gatsby-node.js:309:81
    at step (/Users/hirsch/Development/hackdesign-gatsby/node_modules/gatsby-plugin-excerpts/gatsby-node.js:148:17)
    at Object.next
(/Users/hirsch/Development/hackdesign-gatsby/node_modules/gatsby-plugin-excerpts/gatsby-node.js:79:14)
    at /Users/hirsch/Development/hackdesign-gatsby/node_modules/gatsby-plugin-excerpts/gatsby-node.js:51:67
```

**Comment by YoshiWalsh** [comment]
I have reproduced this issue and will try to work out what causes it and how to fix it.

In the meantime, it appears your use-case is pretty much what [remark-behead](https://www.npmjs.com/package/remark-behead) does. Is it possible possible to use remark plugins with [gatsby-yaml-full-markdown](https://www.gatsbyjs.org/packages/gatsby-yaml-full-markdown/)?

**Comment by YoshiWalsh** [comment]
Fixed the issue, please update to version 1.1.0.

Also I changed your config file a bit based on what I think you're trying to do:

```javascript
    {
      resolve: `gatsby-plugin-excerpts`,
      options: {
        sources: {
          beheadedDescription: {
            type: 'htmlQuery',
            sourceField: 'description',
            excerptSelector: 'body > *',
            elementReplacements: [
              {
                selector: 'h6',
                replaceWith: 'strong'
              },
              {
                selector: 'h5',
                replaceWith: 'h6'
              },
              {
                selector: 'h4',
                replaceWith: 'h5'
              },
              {
                selector: 'h3',
                replaceWith: 'h4'
              },
              {
                selector: 'h2',
                replaceWith: 'h3'
              },
              {
                selector: 'h1',
                replaceWith: 'h2'
              },
            ],
          },
        },
        sourceSets: {
          beheadDescription: [
            `beheadedDescription`
          ]
        },
        excerpts: {
          processedDescription: {
            type: `html`,
            nodeTypeSourceSet: {
              ContentYaml: `beheadDescription`
            }
          }
        }
      },
    }
```

Now in GraphQL Explorer:

```graphql
query MyQuery {
  contentYaml {
    name
    description
    processedDescription
  }
}
```

```json
{
  "data": {
    "contentYaml": {
      "name": "Dropbox",
      "description": "<h1>test</h1>\n<p><strong>Wherever you are</strong><br>\nPut your stuff in Dropbox and get to it from your computers, phones, or tablets. Edit docs, automatically add photos, and show off videos from anywhere.</p>\n",
      "processedDescription": "<h2>test</h2><p><strong>Wherever you are</strong><br>\nPut your stuff in Dropbox and get to it from your computers, phones, or tablets. Edit docs, automatically add photos, and show off videos from anywhere.</p>"
    }
  }
}
```

**Comment by ianhirschfeld** [comment]
Thank you for the update and quick turn around! There is something else that I may have found, but I will put it in a new ticket since it not be related.
